controller-server/src/main/java/com/yahoo/vespa/hosted/controller/endpointcertificates/EndpointCertificateManager.java

```diff
@@ -139,10 +140,13 @@ private boolean verifyEndpointCertificate(EndpointCertificateMetadata endpointCe
                     .filter(san -> san.getType().equals(SubjectAlternativeName.Type.DNS_NAME))
                     .map(SubjectAlternativeName::getValue).collect(Collectors.toSet());
 
-            if (!subjectAlternativeNames.equals(Set.copyOf(dnsNamesOf(instance.id(), List.of(zone)))))
-                return logWarning(warningPrefix, "The list of SANs in the certificate does not match what we expect");
+            if(Sets.intersection(subjectAlternativeNames, Set.copyOf(dnsNamesOf(instance.id(), List.of(zone)))).isEmpty()) {
+                return logWarning(warningPrefix, "No overlap between SANs in certificate and expected SANs");
+            }
 
             return true; // All good then, hopefully
+        } catch (SecretNotFoundException s) {
+            return logWarning(warningPrefix, "Certificate not found in secret store");
```

Comment:
This is expected on first time deployment, but not on the subsequent deployments. Consider not invoking this code path on first time deployment or include a log stating that this can be expected in certain cases. 

On any subsequent calls this should be considered Error, not warning.